From 855f642be539f0468ef6ed8dff3994d4fdf99030 Mon Sep 17 00:00:00 2001
Message-ID: <855f642be539f0468ef6ed8dff3994d4fdf99030.1720090727.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1720090727.git.sharlatanus@gmail.com>
References: <cover.1720090727.git.sharlatanus@gmail.com>
From: Rodion Goritskov <rodion.goritskov@gmail.com>
Date: Wed, 1 May 2024 00:42:43 +0400
Subject: [PATCH v2 3/9] gnu: Add go-github-com-x448-float16.

* gnu/packages/golang.scm (go-github-com-x448-float16): New variable.

Change-Id: I02d8b8b6fea19f6a7b882cfa5d0165d7751f76bf
Signed-off-by: Sharlatan Hellseher <sharlatanus@gmail.com>
---
 gnu/packages/golang-maths.scm | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/gnu/packages/golang-maths.scm b/gnu/packages/golang-maths.scm
index 2348255119..9516f43573 100644
--- a/gnu/packages/golang-maths.scm
+++ b/gnu/packages/golang-maths.scm
@@ -1,4 +1,5 @@
 ;;; GNU Guix --- Functional package management for GNU
+;;; Copyright © 2024 Rodion Goritskov <rodion.goritskov@gmail.com>
 ;;; Copyright © 2024 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;;
 ;;; This file is part of GNU Guix.
@@ -88,6 +89,33 @@ (define-public go-github-com-shopspring-decimal
 @end itemize")
     (license license:expat)))
 
+(define-public go-github-com-x448-float16
+  (package
+    (name "go-github-com-x448-float16")
+    (version "0.8.4")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/x448/float16")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0qg6ya30fra20hpa2qzqqzs8l95lvw9yzd87fdzq195xqi6crb2l"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/x448/float16"))
+    (home-page "https://github.com/x448/float16")
+    (synopsis "Float16 (Binary16) in Go/Golang")
+    (description
+     "Package provides
+@url{https://en.wikipedia.org/wiki/Half-precision_floating-point_format,IEEE
+754 half-precision floating-point format (binary16)} with IEEE 754 default
+rounding for conversions.  IEEE 754-2008 refers to this 16-bit floating-point
+format as binary16.")
+    (license license:expat)))
+
 ;;;
 ;;; Executables:
 ;;;
-- 
2.41.0

